src/shillelagh/adapters/api/gsheets/adapter.py

```diff
@@ -462,6 +468,27 @@ def insert_data(self, row: Row) -> int:
 
         return row_id
 
+    def _check_permissions(self) -> None:
+        """
+        Check if we have permission to access a sheet.
+        """
+        session = self._get_session()
+        url = (
+            f"https://sheets.googleapis.com/v4/spreadsheets/{self._spreadsheet_id}"
+            f"/developerMetadata/{self._sheet_id}"
+        )
+        response = session.get(url)
+        if not response.ok:
+            payload = response.json()
+            error = payload["error"]
+
+            # custom exception to trigger oauth
+            if error["code"] == 401:
+                raise UnauthenticatedError(error["message"])
+
+            # something else happened, raise exception with the message
+            raise InterfaceError(error["message"])
```

Comment:
Perhaps raise from the original exception?

Reply:
Ah, thanks for the comment! I thought of doing that, but it would require passing the original exception to `_check_permissions`, which felt a bit weird. But I guess it would be good, since this exception will be unrelated to the original one that triggered this part of the code, so it's good to have the original information. I'll add it in my next PR, thanks!